### PR TITLE
Sigint 725 - cache download

### PR DESCRIPTION
**Pull request by lsynopsys**

### CODE REVIEW

File: synopsys-task/test/unit/synopsys-task/synopsys-bridge.spec.ts [hunk 1]
```diff
@@ -102,4 +111,468 @@ describe("Synopsys Bridge test", () => {
...
+        it("throws an error when BRIDGE_DOWNLOAD_URL is defined but invalid", async () => {
+            Object.defineProperty(inputs, "BRIDGE_DOWNLOAD_URL", {
+                value: "invalid-url",
+            });
+            synopsysBridge.getBridgeUrl().catch(errorObj => {
+                expect(errorObj.message).includes("Invalid URL")
+            })
+
+            Object.defineProperty(inputs, "BRIDGE_DOWNLOAD_URL", {
+                value: undefined,
```
Comment (lsynopsys) [inline synopsys-task/test/unit/synopsys-task/synopsys-bridge.spec.ts:245]: add empty instead undefined

File: synopsys-task/test/unit/synopsys-task/synopsys-bridge.spec.ts [hunk 2]
```diff
@@ -102,4 +111,468 @@ describe("Synopsys Bridge test", () => {
...
+            });
+        });
+
+        it("returns the URL for the specified version when BRIDGE_DOWNLOAD_VERSION is defined and valid", async () => {
+            Object.defineProperty(inputs, "BRIDGE_DOWNLOAD_VERSION", {
+                value: "0.1.244",
+            });
+            sandbox.stub(synopsysBridge, "validateBridgeVersion").returns(Promise.resolve(true));
+            //sandbox.stub(synopsysBridge, "getVersionUrl").returns(bridgeUrl);
```
Comment (lsynopsys) [inline synopsys-task/test/unit/synopsys-task/synopsys-bridge.spec.ts:254]: uncommment

File: synopsys-task/src/synopsys-task/utility.ts [hunk 3]
```diff
@@ -88,3 +88,10 @@ export function getWorkSpaceDirectory(): string {
     throw new Error("Workspace directory could not be located");
   }
 }
+
+export function checkIfPathExists(fileOrDirectoryPath: string): boolean {
```
Comment (lsynopsys) [inline synopsys-task/src/synopsys-task/utility.ts:92]: use tasklib method

File: synopsys-task/task.json [hunk 4]
```diff
@@ -6,11 +6,11 @@
   "description": "Downloads specified executor and executes specified command",
   "helpMarkDown": "",
   "category": "Utility",
-  "author": "spurohit",
+  "author": "lsynopsys",
   "version": {
     "Major": 0,
     "Minor": 1,
-    "Patch": 2
+    "Patch": 5
```
Comment (kishorikumar) [inline synopsys-task/task.json:13]: Change it to 0. Let's have the first release version in our main branch.

File: vss-extension.json [hunk 5]
```diff
@@ -2,7 +2,7 @@
   "manifestVersion": 1,
   "id": "synopsys-extension",
   "name": "synopsys-extension",
-  "version": "0.1.0",
+  "version": "0.1.5",
```
Comment (kishorikumar) [inline vss-extension.json:5]: Change it to 0. Let's have the first release version in our main branch.

**Review by kishorikumar — CHANGES_REQUESTED** [review]

**Comment by kishorikumar** [comment]
Make sure all the generated files inside dist directory are also pushed to main branch

**Review by kishorikumar — APPROVED** [review]

**Pull request merged**